### PR TITLE
Simplify state machine of pending summary state after summary is submitted

### PR DESCRIPTION
**Pull request by NicholasCouri**
The initial goal for this task was to simplify the state machine of the pending summary state after a summary is submitted. 

In order to better understand how often the scenario happens, we are simply adding a telemetry event that will be logged whenever the proposal handle is not on the pendingsummaries list. 

Addressing part of [#AB444](https://dev.azure.com/fluidframework/internal/_workitems/edit/444)

Note: this will also help understand a problem on FRS in which the pendingsummaries list never seems to have the proposal handle.

### CODE REVIEW

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 1]
```diff
@@ -224,6 +224,14 @@ export class SummarizerNode implements IRootSummarizerNode {
                 this.refreshLatestSummaryFromPending(proposalHandle, maybeSummaryNode.referenceSequenceNumber);
                 return { latestSummaryUpdated: true, wasSummaryTracked: true };
             }
+
+            this.defaultLogger.sendTelemetryEvent({
+                eventName: "PendingSummaryNotFound",
+                proposalHandle,
+                summaryRefSeq,
+                size: this.pendingSummaries.size,
+                haveHandle: this.pendingSummaries.size > 0 ? this.pendingSummaries.keys[0] : undefined,
```
Comment (andre4i) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:233]: What does `haveHandle` represent?

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts [hunk 2]
```diff
@@ -224,6 +224,14 @@ export class SummarizerNode implements IRootSummarizerNode {
                 this.refreshLatestSummaryFromPending(proposalHandle, maybeSummaryNode.referenceSequenceNumber);
                 return { latestSummaryUpdated: true, wasSummaryTracked: true };
             }
+
+            this.defaultLogger.sendTelemetryEvent({
+                eventName: "PendingSummaryNotFound",
+                proposalHandle,
+                summaryRefSeq,
+                size: this.pendingSummaries.size,
```
Comment (andre4i) [inline packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:232]: I don't have a strong opinion on this, but most telemetry use `length` as the property name for similar stuff

**Review by andre4i — APPROVED** [review]

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts (hunk 2 above)
Comment (NicholasCouri) [reply packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:232]: Same here - I saw there was an existing field called size and I used it. I will change to length if it is more used.

File: packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts (hunk 1 above)
Comment (NicholasCouri) [reply packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts:233]: HaveHandle is an existing column I'm making use of to identify the proposal we have pendingsummary for (local list).
I'm also adding the referenceSequenceNumber from that proposal and the current one. Let me know what you think :)

**Review by tyler-cai-microsoft — APPROVED** [review]

**Pull request merged**